### PR TITLE
Add Jaguar commands

### PR DESCRIPTION
**Pull request by lask**

### CODE REVIEW

File: vscode/package.json [hunk 1]
```diff
@@ -131,6 +134,41 @@
 				"category": "Toit",
 				"icon": "$(selection)",
 				"enablement": "toit.extensionActive"
+			},
+			{
+				"command": "jag.watch",
+				"title": "Watch for changes and rerun",
+				"category": "Jag",
```
Comment (kasperl) [inline vscode/package.json:141]: Maybe make the category 'Jaguar'?

File: vscode/package.json [hunk 2]
```diff
@@ -225,18 +263,26 @@
...
+					"description": "The path to the 'toit' executable, e.g. `/usr/bin/toit`.",
 					"deprecationMessage": "Deprecated: Use toit.path instead.",
 					"scope": "machine-overridable"
 				},
+				"jag.path": {
+					"type": [
+						"string",
+						"null"
+					],
+					"description": "The path to the 'jag' executable, e.g. `/usr/bin/jsg`. If left unset, the 'jsg' executable in the path is used.",
```
Comment (kasperl) [inline vscode/package.json:283]: jsg -> jag

File: vscode/src/extension.ts [hunk 3]
```diff
@@ -172,6 +189,14 @@ async function findExecutables(): Promise<Executables> {
     lspCommand = configLspCommand;
   }
 
+  if (jagExec === null) {
+    const jagResult = run("jag", ["version"]);
```
Comment (kasperl) [inline vscode/src/extension.ts:193]: You might be able to use 'jag setup --check' by now.

File: vscode/src/jagCmds.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      "canSelectMany": false,
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      "filters": {"Toit": ["toit"]},
+      "title": "Select Toit-file to watch"
+    });
+
+    if (!filePath) return;
+
+    const prefix = ctx.watchNumber === 0 ? "" : ` (${ctx.watchNumber++})`;
+    const terminal = Window.createTerminal(`Jag watch${prefix}`);
```
Comment (kasperl) [inline vscode/src/jagCmds.ts:18]: Jag -> Jaguar

or

Jag -> jag

File: vscode/src/jagCmds.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+async function executeJagMonitor(ctx: JagContext) {
+  try {
+    const portPromptOptions: InputBoxOptions = {
+      "prompt": "Enter port (optional)"
+    };
+    const port = await Window.showInputBox(portPromptOptions);
+    const portFlag = port ? `--port '${port}'`: "";
+
+    const prefix = ctx.monitorNumber === 0 ? "" : ` (${ctx.monitorNumber++})`;
+    const terminal = Window.createTerminal(`Jag monitor${prefix}`);
```
Comment (kasperl) [inline vscode/src/jagCmds.ts:54]: Jag -> Jaguar

or

Jag -> jag

File: vscode/src/utils.ts [hunk 6]
```diff
@@ -12,6 +12,25 @@ import { updateStatus } from "./projectCmd";
...
+  sharedTerminal: Terminal|null = null;
+  jagExec: string;
+
+  constructor(jagExec: string) {
+    this.jagExec = jagExec;
+  }
+
+  ensureSharedTerminal() : Terminal {
+    if (!this.sharedTerminal || this.sharedTerminal.exitStatus) {
+      this.sharedTerminal = Window.createTerminal(`Jag`);
```
Comment (kasperl) [inline vscode/src/utils.ts:28]: Jag -> Jaguar

or

Jag -> jag

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: vscode/src/extension.ts (hunk 3 above)
Comment (lask) [reply vscode/src/extension.ts:193]: Will look at this in a different PR.

File: vscode/package.json [hunk 7]
```diff
@@ -225,18 +263,26 @@
 						"null"
 					],
 					"default": null,
-					"description": "The path to 'toit' executable, e.g. `/usr/bin/toit`. If left unset, the 'toit' executable in the path is used. If set to empty, then the extension only uses the language server.",
+					"description": "The path to the 'toit' executable, e.g. `/usr/bin/toit`. If left unset, the 'toit' executable in the path is used. If set to empty, then the extension only uses the language server.",
```
Comment (floitsch) [inline vscode/package.json:266]: The second part of the description isn't really true anymore. I would just remove it.

**Review by floitsch — COMMENTED** [review]
LGTM.
Sorry for the delay. I didn't see the review request.

File: vscode/package.json (hunk 7 above)
Comment (lask) [reply vscode/package.json:266]: I will fix it!

**Pull request merged**